### PR TITLE
Autobuild Function calling

### PR DESCRIPTION
**Pull request by krishnashed**
## Why are these changes needed?

Introduction of Autobuild, didn't support function/tool calling. This PR is solution to add this feature/

## Related issue number

Closes https://github.com/microsoft/autogen/issues/1510

### CODE REVIEW

File: test/agentchat/contrib/test_agent_builder.py [hunk 1]
```diff
@@ -69,9 +92,111 @@ def test_build():
...
+
+        # Verify that 'ask_ossinsight' was called exactly once during the task execution.
+        mocked_function.assert_called()
+
+
 @pytest.mark.skipif(
-    skip_openai or skip,
-    reason=reason + "OR dependency not installed",
+    skip,
+    reason="requested to skip",
```
Comment (sonichi) [inline test/agentchat/contrib/test_agent_builder.py:199]: What's the reason of modifying the skip condition?

File: test/agentchat/contrib/test_agent_builder.py [hunk 2]
```diff
@@ -69,9 +92,111 @@ def test_build():
...
+            execution_task="How many stars microsoft/autogen has on GitHub?",
+            agent_list=agent_list,
+        )
+
+        # Verify that 'ask_ossinsight' was called exactly once during the task execution.
+        mocked_function.assert_called()
+
+
+@pytest.mark.skipif(
+    skip,
```
Comment (sonichi) [inline test/agentchat/contrib/test_agent_builder.py:140]: Should the condition be `skip_openai or skip` ?

File: test/agentchat/contrib/test_agent_builder.py [hunk 3]
```diff
@@ -1,17 +1,22 @@
 #!/usr/bin/env python3 -m pytest
 
+#!/usr/bin/env python3 -m pytest
+
```
Comment (Hk669) [inline test/agentchat/contrib/test_agent_builder.py:4]: we can remove this duplicate shebang

**Comment by Hk669** [comment]
#3235 is it a duplicate or a new PR extending this PR? @krishnashed

**Comment by LinxinS97** [comment]
> #3235 is it a duplicate or a new PR extending this PR? @krishnashed 

Openai test cannot run on previous PR and I will close that one.

**Pull request closed without merging**